lib/r10k/module/base.rb

```diff
@@ -52,6 +52,7 @@ def initialize(title, dirname, args, environment=nil)
     @path = Pathname.new(File.join(@dirname, @name))
     @environment = environment
     @overrides = args.delete(:overrides) || {}
+    @deploy_spec = @overrides.dig(:modules, :deploy_spec) || args.delete(:deploy_spec)
```

Comment:
If we're deleting `deploy_spec` from the hash here, will it work with `setopts` in the subclasses? It looks like that will just [iterate](https://github.com/puppetlabs/r10k/blob/ded1045f29f99a624c19eae25d96217cd0dee179/lib/r10k/util/setopts.rb#L40) over what's in the hash and do things with each key, but if the key has been deleted when we get there, I think that will always be a no-op for `deploy_spec`. 

I think what you have here might be sufficient to make it available as an instance var in the subclasses without adding it to `setopts`? But I might be misunderstanding something.

Reply:
ahh yeah, there's a bug here where we don't delete it from the overrides if it exists from the cli, I knew there was something fishy about me adding it to all the `setopts` in the subclasses. I'll change it to delete it from the overrides, and then maybe we can just avoid this setopts business altogether...unless we think there is a good reason to maintain it in `args`? We are already mutating args in this `initialize`, so it seems fine to do what I just described...

Reply:
Agreed, let's just delete it and handle this all in the base class.